app/services/local_auth_client.rb

```
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class LocalAuthClient
+  NAME = "SparkJoy Local Client"
+  REDIRECT_URI = "https://sparkjoy.com/oauth"
+  SCOPES = "public"
+  TIME_UNTIL_EXPIRATION = Doorkeeper.configuration.access_token_expires_in
+
+  def self.create_token(user)
+    new(user).create_token
+  end
+
+  def initialize(user)
+    @user_id = user.id
+  end
+
+  def create_token
+    Doorkeeper::AccessToken.create(
```

Is `Doorkeep::AccessToken` an ActiveRecord model? If so, is it OK if this silently fails to persist, or would it be better to raise an error with `create!`?

---

It is a model. Good point.

---

A little more nuance. For this use, in the service, it makes sense to fail loudly. In the mutation, since we're using this pattern for [errors as data](https://graphql-ruby.org/mutations/mutation_errors#errors-as-data), we don't want to fail loudly, and instead return the errors to the client.